### PR TITLE
Either right if null extension

### PR DESCRIPTION
**Pull request by Neuman968**
Added rightIfNull Either extension function for kotlin.Any? type. Provides inverse functionality of rightIfNotNull.

### CODE REVIEW

File: modules/core/arrow-core-data/src/main/kotlin/arrow/core/Either.kt [hunk 1]
```diff
@@ -382,6 +382,20 @@ fun <A, B> B?.rightIfNotNull(default: () -> A): Either<A, B> = when (this) {
...
 }
 
+/**
+ * Returns [Either.Right] if the value of type Any? is null, otherwise the specified A value wrapped into an
+ * [Either.Left].
+ *
+ * Example:
+ * ```
+ * null.rightIfNull { "left" }    // Right(b="Unit")
+ * "value".rightIfNull { "left" ) // Left(a="left")
```
Comment (pakoito) [inline modules/core/arrow-core-data/src/main/kotlin/arrow/core/Either.kt]: There's a mismatched `{`

**Comment by pakoito** [comment]
Nice, very complete PR. Thank you :D

I would need an example of where `Unit` in the right is the right approach. Wouldn't you want `Nothing?` instead?

**Comment by Neuman968** [comment]
@pakoito  Thank you! This is my first public PR so forgive me if I miss any formalities. 
I did consider using `Nothing?` but from my understanding Nothing cannot be instantiated so the code

```kotlin
fun <A> Any?.rightIfNull(default: () -> A): Either<A, Nothing?> = when (this) {
  null -> Either.right(Nothing())
  else -> Either.left(default())
}
```
Doesn't compile. Maybe I am missing something. I have a few projects that use this library and found a few places where I used this rightIfNull functionality so I thought it would be a good first contribution. 

Thanks for such an awesome project! and Thank you so much for your time :)

File: modules/core/arrow-core-data/src/main/kotlin/arrow/core/Either.kt (hunk 1 above)
Comment (Neuman968) [reply modules/core/arrow-core-data/src/main/kotlin/arrow/core/Either.kt]: Thank you! Fixed.

**Comment by Neuman968** [comment]
On second thought, It could be something like 

```kotlin
fun <A> Any?.rightIfNull(default: () -> A): Either<A, Nothing?> = when (this) {
  null -> Either.right(null)
  else -> Either.left(default())
}
```
I updated the code and README.

File: modules/docs/arrow-docs/docs/docs/arrow/core/either/README.md [hunk 2]
```diff
@@ -317,6 +317,21 @@ Example:
...
 ```
 
+The inverse of `rightIfNotNull`, `rightIfNull`.
+If the value is null it will be transformed to the specified `Either.right` and the type will be `Nothing?`.
+If the value is not null than it will be transformed to the specified `Either.Left`.
+
+Example:
+
+```kotlin:ank
+// Results in Either.Left
```
Comment (pakoito) [inline modules/docs/arrow-docs/docs/docs/arrow/core/either/README.md]: These aren't necessary, ank will display the result inlined below this snippet

File: modules/docs/arrow-docs/docs/docs/arrow/core/either/README.md [hunk 3]
```diff
@@ -317,6 +317,21 @@ Example:
...
+"value".rightIfNull { "left" }
+```
+
+```kotlin:ank
+// Results in Either.Right
```
Comment (pakoito) [inline modules/docs/arrow-docs/docs/docs/arrow/core/either/README.md]: Same here

**Comment by pakoito** [comment]
You're doing okay, no formalities are necessary :D As soon as things are fixed and CI passes we're good to go.

File: modules/docs/arrow-docs/docs/docs/arrow/core/either/README.md (hunk 2 above)
Comment (Neuman968) [reply modules/docs/arrow-docs/docs/docs/arrow/core/either/README.md]: Thanks! should be corrected now :)

**Comment by pakoito** [comment]
Passing and merged. Thank you for the contribution!

**Comment by Neuman968** [comment]
Thank you!

**Pull request merged**